File: grok.cpp
#include <iostream>
#include <chrono>

int main() {
    using namespace std;
    using namespace std::chrono;

    auto start = high_resolution_clock::now();

    char baseline[30];
    for (int i = 0; i < 30; i++)
    baseline[i] = '8';

    int toChange[14][20] = {
        {
            1,
            -1
        },
        {
            -1
        },
        {
            13,
            14,
            15,
            16,
            24,
            25,
            -1
        },
        {
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            22,
            23,
            24,
            -1
        },
        {
            7,
            8,
            9,
            10,
            11,
            12,
            20,
            21,
            22,
            23,
            -1
        },
        {
            6,
            7,
            8,
            9,
            18,
            19,
            20,
            21,
            22,
            23,
            24,
            -1
        },
        {
            6,
            7,
            8,
            16,
            17,
            18,
            21,
            22,
            23,
            -1
        },
        {
            6,
            7,
            8,
            14,
            15,
            16,
            22,
            23,
            24,
            -1
        },
        {
            6,
            7,
            8,
            9,
            21,
            22,
            23,
            -1
        },
        {
            7,
            8,
            9,
            18,
            19,
            20,
            21,
            22,
            23,
            -1
        },
        {
            6,
            7,
            8,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            -1
        },
        {
            4,
            5,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            -1
        },
        {
            -1
        },
        {
            29,
            -1
        }
    };
    for (int t = 0; t < 1000000; t++) {
        for (int i = 0; i < 14; i++) {
            char thisLine[30];

            for (int j = 0; j < 30; j++)
            thisLine[j] = baseline[j];

            for (int k = 0; toChange[i][k] != -1; k++) {
                int index = toChange[i][k];
                thisLine[index] = (i == 0) ? '0': '1';
            }

        }
    }

    auto end = high_resolution_clock::now();
    auto micros = duration_cast < microseconds > (end - start).count();

    cout << "Time taken in microseconds: " << micros << "\n";
    cout << "Time taken in milliseconds: " << micros / 1000.0 << "\n";
    cout << "Time taken in seconds: " << micros / 1000000.0 << "\n";

    return 0;
}

File: grok.rs
use std::time::Instant;

fn main() {
    println!("Hello from RUST");
    let start = Instant::now();

    let baseline: [&str; 30] = ["8"; 30];

    let to_change: [&[i32]; 14] = [
        &[1],
        &[],
        &[13, 14, 15, 16, 24, 25],
        &[9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 22, 23, 24],
        &[7, 8, 9, 10, 11, 12, 20, 21, 22, 23],
        &[6, 7, 8, 9, 18, 19, 20, 21, 22, 23, 24],
        &[6, 7, 8, 16, 17, 18, 21, 22, 23],
        &[6, 7, 8, 14, 15, 16, 22, 23, 24],
        &[6, 7, 8, 9, 21, 22, 23],
        &[7, 8, 9, 18, 19, 20, 21, 22, 23],
        &[6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21],
        &[4, 5, 11, 12, 13, 14, 15, 16, 17, 18],
        &[],
        &[-1],
    ];


for _ in 0..1_000_000 {
    for i in 0..14 {

        let mut output = baseline;
        
        for &index in to_change[i] {
            if index >= 0 {
                output[index as usize] = if i == 0 { "0" } else { "1" };
            }
        }

    }
    }

    let duration = start.elapsed();
    let micros = duration.as_micros();
    let _millis = duration.as_millis();
    let secs = duration.as_secs_f64();

    println!("Time taken in microseconds: {}", micros);
    println!("Time taken in milliseconds: {}", micros as f32/1000.0);
    println!("Time taken in seconds: {:.6}", secs);
}
